virtual_rainforest/models/animals/animal_model.py

```diff
@@ -0,0 +1,112 @@
+"""The :mod:`~virtual_rainforest.models.animal.animal_model` module creates a
+:class:`~virtual_rainforest.models.animal.animal_model.AnimalModel` class as a
+child of the :class:`~virtual_rainforest.core.model.BaseModel` class.
+At present a lot of the abstract methods of the parent class (e.g.
+:func:`~virtual_rainforest.core.model.BaseModel.setup` and
+:func:`~virtual_rainforest.core.model.BaseModel.spinup`) are overwritten using
+placeholder functions that don't do anything. This will change as the
+:mod:`virtual_rainforest` model develops. The factory method
+:func:`~virtual_rainforest.models.animal.animal_model.AnimalModel.from_config`
+exists in a more complete state, and unpacks a small number of parameters
+from our currently pretty minimal configuration dictionary. These parameters are
+then used to generate a class instance. If errors crop up here when converting the
+information from the config dictionary to the required types
+(e.g. :class:`~numpy.timedelta64`) they are caught and then logged, and at the end
+of the unpacking an error is thrown. This error should be caught and handled
+by downstream functions so that all model configuration failures can be reported as one.
+"""  # noqa: D205, D415
+
+
+from __future__ import annotations
+
+from typing import Any
+
+import pint
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.base_model import BaseModel, InitialisationError
+from virtual_rainforest.core.logger import LOGGER
+
+
+class AnimalModel(BaseModel):
+    """A class describing the animal model.
+
+    Describes the specific functions and attributes that the animal module should
+    possess. Currently it is incomplete and mostly just a copy of the template set out
+    in AnimalModel.
+
+    Args:
+        update_interval: Time to wait between updates of the model state.
+        start_time: Time at which the model is initialized.
+    """
+
+    model_name = "animal"
+    """The model name for use in registering the model and logging."""
+
+    def __init__(
+        self,
+        update_interval: timedelta64,
```

Comment:
This now needs to include `data` - my PR just got merged... See the updated docs: 

https://virtual-rainforest.readthedocs.io/en/latest/development/defining_new_models.html#defining-the-model-init-method

Reply:
Are the docs for from_config up to date with the inclusion of `data`?

Reply:
Actually, the broader question I have here is about reworking the testing file for animal_model. (I sent you a teams message but it probably makes more sense on here). 

Is there a specific way I should be generating a dummy data object for use in testing, if I am not actually using data for testing?
Would declaring, 'grid = Grid(cell_nx=3, cell_ny=1)     data = Data(grid)' work?

Reply:
That's a good question, and I had forgotten about updating the tests. There is a shared fixture for just this purpose included in `conftest.py`: `data_instance`.

So, your test functions can now use that fixture name as an argument and get that shared `Data` instance:

```python
def test_the_thing(..., data_instance, ...):

    AnimalModel(data_instance, ...)
```

Reply:
If there is a need to have that shared data instance do something slightly different, we can chat about what each model needs.

Reply:
Great, thanks!